### ISSUE TITLE
Compute Instance naming rule is not identical with Azure ML portal

### ISSUE THREAD

**Issue by AngieXiong** [hashicorp/terraform-provider-azurerm#15364]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
The maximum length of a compute instance name is 16 characters, while its length at Azure portal is 28 characters.

### Potential Terraform Configuration
N/A

### Affected Resource(s)

* `azurerm_machine_learning_compute_instance`

### Expected Behaviour
When the length of compute instance is between 16 and 28, the instance should be successfully created.

### Actual Behaviour
An error shows:
```hcl
Error: invalid value for name (It can include letters, digits and dashes. It must start with a letter, end with a letter or digit, and be between 2 and 16 characters in length.)
```

### References

N/A

**Comment by ms-henglu** [comment]
@AngieXiong , thank you for taking time to report this issue.

The validation has been updated and released in v2.97.0, would you please have a try? Thanks!

**Comment by stephybun** [comment]
Closed by #15402
